draco/analysis/transform.py

```diff
@@ -552,6 +552,59 @@ def _pack_marray(mmodes, mmax=None):
     return marray
 
 
+class MModeInverseTransform(task.SingleTask):
+    def process(self, ma):
+
+        ma.redistribute("freq")
+        ntime = ma.vis.shape[0] * 2 - 1
+
+        # Re-construct array of S-streams
+        ssarray = _make_ssarray(ma.vis[:])  # , mmax)
+        ssarray = mpiarray.MPIArray.wrap(ssarray[:], axis=0, comm=ma.comm)
+
+        # Construct container and set visibility data
+        sstream = containers.SiderealStream(
+            ra=ntime, axes_from=ma, distributed=True, comm=ma.comm
+        )
+        sstream.redistribute("freq")
+
+        # Assign the visibilities and weights into the container
+        sstream.vis[:] = ssarray
+        # There is no way to recover time information for the weights.
+        # Just assign the time average to each baseline and frequency.
+        sstream.weight[:] = ma.weight[0, 0, :, :][:, :, np.newaxis] / float(ntime)
+
+        return sstream
+
+
+def _make_ssarray(ma):  # , mmax):
+    # Construct an array of sidereal time streams from m-modes
+    #
+    marray = _unpack_marray(ma)
+    ssarray = np.fft.ifft(marray * marray.shape[-1], axis=-1)
```

Comment:
Sorry I got confused in my previous comment. But what I'm wondering about is do we need to alter the normalisation if the number of samples is different in the inverse transform? (or if some modes were cut)

Reply:
I have tested the new code and the normalization works fine now.